updatemetadata.php

```diff
@@ -0,0 +1,93 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ *
+ * * @package    block_opencast
+ * @copyright  2019 Farbod Zamani, ELAN e.V.
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+require_once('../../config.php');
+
+use block_opencast\local\upload_helper;
+
+global $PAGE, $OUTPUT, $CFG;
+
+require_once($CFG->dirroot . '/repository/lib.php');
+
+$identifier = required_param('video_identifier', PARAM_ALPHANUMEXT);
+$courseid = required_param('courseid', PARAM_INT);
+
+$baseurl = new moodle_url('/blocks/opencast/updatemetadata.php',  array('identifier' => $identifier, 'courseid' => $courseid));
+$PAGE->set_url($baseurl);
+
+$redirecturl = new moodle_url('/blocks/opencast/index.php', array('courseid' => $courseid));
+
+require_login($courseid, false);
+
+$PAGE->set_pagelayout('incourse');
+$PAGE->set_title(get_string('pluginname', 'block_opencast'));
+$PAGE->set_heading(get_string('pluginname', 'block_opencast'));
+$PAGE->navbar->add(get_string('pluginname', 'block_opencast'), $redirecturl);
+$PAGE->navbar->add(get_string('updatemetadata', 'block_opencast'), $baseurl);
+
+// Capability check.
+$coursecontext = context_course::instance($courseid);
+require_capability('block/opencast:addvideo', $coursecontext);
+
+$opencast = \block_opencast\local\apibridge::get_instance();
+$metadata = $opencast->get_event_metadata($identifier, '?type=dublincore/episode');
+$languages = upload_helper::get_opencast_metadata_catalog(['name' => 'language'], 'param_json');
```

Comment:
Again the question why these two metadata fields are hardcoded. Why not use the same dymanic method as for the addvideo_form?

Reply:
The reason is to consider that, admin did not set the values via admin page and there are some uploaded tasks that needed to be updated in advanced! However, it tries to get its values dynamically as it defined in upload_helper::get_opencast_metadata_catalog(), in case there is nothing defined it uses the hardcoded values!

Reply:
> there are some uploaded tasks that needed to be updated in advanced
Do you mean at the point in time, when an admin installs this patch on their system and there are still upload_tasks to be processed? Or the case, when the admin changes the catalog in the adminsetting?

Reply:
The first scenario has been considered mainly

Reply:
Why don't we simply process existing upload_jobs the same way, as we did before the patch? For this purpose, we only need to extend apibridge to add the filename as a title again in case the metadata of the job is empty.

Reply:
Then values for metadata fields, which are not defined by the admin, just remain empty. Before this patch, we did not set the language at all. Also the admin would have to inform the teachers to alter the metadata after the patch for existing uploads. I think this will not happen. At least not before the admin specified the metadata catalog as intended.

Reply:
I think there is a misunderstanding about the functionality of the updatemetadata, which is intended to help users edit their predefined metadata fields (in Add Video) and it is not mandatory. Additionally, there is no need to inform teachers to alter the metadata and there is no conflict with the uploaded jobs before this patch.

Reply:
However, I will remove the hardcoded values and variables and try to reform it in the way that it gets the fields from metadata catalog defined by admin and its values via the API call from Opencast.